### PR TITLE
fix Issue 20898 - order dependency in evaluating pragma(inline) for f…

### PR DESCRIPTION
**Pull request by WalterBright**
…unctions

### CODE REVIEW

**Comment by WalterBright** [comment]
This may be impractical.

File: src/dmd/statementsem.d [hunk 1]
```diff
@@ -2437,13 +2437,19 @@ else
...
-                    ps.error("`pragma(inline)` is not inside a function");
-                    return setError();
+                    /* Allow only if semantic analysis will be done before function is called
+                     */
+                    if (!fd.isFuncLiteralDeclaration() &&
+                        !fd.inferRetType &&
+                        !fd.toParent2().isFuncDeclaration() &&
+                        !fd.isInstantiated())
+                    {
+                        ps.deprecation("move `pragma(inline)` outside of function `%s`", fd.toChars());
```
Comment (UplinkCoder) [inline src/dmd/statementsem.d:2449]: function body?

**Comment by WalterBright** [comment]
I'm giving up on this due to too much code breakage.

**Comment by UplinkCoder** [comment]
Thanks that's a relieve.

**Comment by dnadlinger** [comment]
I (still) think this should be deprecated regardless. It should be possible to skip parsing imported function bodies entirely, so always-inline information belongs on the declaration.

**Pull request closed without merging**